### ISSUE TITLE
Unable to run Spoke after install on Win10 x64 - ERR! Failed at the spoke@2.0.0 start script

### ISSUE THREAD

**Issue by groff0nthewall** [ProgressiveCoders/Spoke#1137]
Followed the Getting Started instructions but running `yarn dev` gives me the error:
`ERR! Failed at the spoke@2.0.0 start script`
and I just don't know how to get around it. The package.json start script references a path that doesn't appear to exist: `"start": "node ./build/server/server"`

NOTE: when following the Getting Started instructions I initially got an error installing sqlite that I _thought was resolved_ by following the suggestions in this article: https://github.com/mapbox/node-sqlite3/issues/612. 
I'm not sure if this is related to my current issue, but mention it just in case it is helpful info.

**How can I resolve this error so that I may build/run locally in my Win 10 environment?**

Full console output and error message:
```c:\source\myProj-2>yarn dev
yarn run v1.16.0
$ nf start -w --procfile ./dev-tools/Procfile.dev
[OKAY] Loaded ENV .env File as KEY=VALUE Format
[WARN] No Procfile Found
[OKAY] package.json file found - trying 'npm start'
[OKAY] Wrapping display Output to 216 Columns
10:49:57 AM web.1 |  > spoke@2.0.0 start c:\source\myProj-2
10:49:57 AM web.1 |  > node ./build/server/server
10:49:57 AM web.1 |  internal/modules/cjs/loader.js:582
10:49:57 AM web.1 |      throw err;
10:49:57 AM web.1 |      ^
10:49:57 AM web.1 |  Error: Cannot find module 'c:\source\myProj-2\build\server\server'
10:49:57 AM web.1 |      at Function.Module._resolveFilename (internal/modules/cjs/loader.js:580:15)
10:49:57 AM web.1 |      at Function.Module._load (internal/modules/cjs/loader.js:506:25)
10:49:57 AM web.1 |      at Function.Module.runMain (internal/modules/cjs/loader.js:741:12)
10:49:57 AM web.1 |      at startup (internal/bootstrap/node.js:285:19)
10:49:57 AM web.1 |      at bootstrapNodeJSCore (internal/bootstrap/node.js:739:3)
10:49:57 AM web.1 |  npm
10:49:57 AM web.1 |   ERR! code ELIFECYCLE
10:49:57 AM web.1 |  npm ERR! errno 1
10:49:57 AM web.1 |  npm ERR! spoke@2.0.0 start: `node ./build/server/server`
10:49:57 AM web.1 |  npm ERR! Exit status 1
10:49:57 AM web.1 |  npm ERR!
10:49:57 AM web.1 |  npm ERR! Failed at the spoke@2.0.0 start script.
10:49:57 AM web.1 |  npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
10:49:57 AM web.1 |  npm ERR! A complete log of this run can be found in:
10:49:57 AM web.1 |  npm ERR!     C:\Users\user_name\AppData\Roaming\npm-cache\_logs\2019-06-11T14_49_57_363Z-debug.log
[DONE] Killing all processes with signal  null
internal/util.js:229
  throw new ERR_UNKNOWN_SIGNAL(signal);
  ^

TypeError [ERR_UNKNOWN_SIGNAL]: Unknown signal: null
    at convertToValidSignal (internal/util.js:229:9)
    at ChildProcess.kill (internal/child_process.js:414:5)
    at EventEmitter.<anonymous> (c:\source\myProj-2\node_modules\foreman\lib\proc.js:49:11)
    at EventEmitter.emit (events.js:187:15)
    at ChildProcess.<anonymous> (c:\source\myProj-2\node_modules\foreman\lib\proc.js:45:13)
    at ChildProcess.emit (events.js:182:13)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.```

**Comment by joshco** [comment]
What shell are you using?  Is that git bash?  You might be running into path conversion issues.

You might have better luck using Ubuntu for Windows
https://www.microsoft.com/en-us/p/ubuntu/9nblggh4msv6?activetab=pivot:overviewtab
That's a full ubuntu running on  top of the windows kernel.  I have found that my dev environments are much better this way because it is pretty much identical to a real linux install.

**Comment by joshco** [comment]
Also, try changing Procfile.dev from CRLF to LF line separators.  dos2unix is one tool.  I just ran into that

**Comment by ibrand** [comment]
I encountered the same error, and just put in a PR to try and do a fix on the repo level for it. @joshco was right in that I was able to solve it by converting my linefeeds to LF. Thanks Josh!

@groff0nthewall if you'd be up for it, you could clone my branch and let me know if it also fixes your problem by doing:
`git clone -b fix-linefeed-issue git@github.com:MoveOnOrg/Spoke.git`
and then stepping through [steps 2, 3, and 4 from the README](https://github.com/MoveOnOrg/Spoke/blob/main/README.md) on the new repo and running `yarn dev` to see if it works!

If it does, let me know so I can comment on it and get it merged into the main branch so that you can use that. Otherwise, happy to try and debug further with you.

**Comment by groff0nthewall** [comment]
> What shell are you using? Is that git bash? You might be running into path conversion issues.
> 
> You might have better luck using Ubuntu for Windows
> https://www.microsoft.com/en-us/p/ubuntu/9nblggh4msv6?activetab=pivot:overviewtab
> That's a full ubuntu running on top of the windows kernel. I have found that my dev environments are much better this way because it is pretty much identical to a real linux install.

Using windows cmd prompt (as admin), but just tried git bash and got same result.

**Comment by groff0nthewall** [comment]
Converted Procfile.dev using dos2unix as suggested by @joshco but that didn't solve. Got the following error:

```c:\proj\SEIU\seiu-2>yarn dev-start
yarn run v1.16.0
$ npm run clean

> spoke@2.0.0 clean c:\proj\SEIU\seiu-2
> rm -rf $OUTPUT_DIR

'rm' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! spoke@2.0.0 clean: 'rm -rf $OUTPUT_DIR'
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the spoke@2.0.0 clean script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\MyUserName\AppData\Roaming\npm-cache\_logs\2019-06-27T14_44_07_046Z-debug.log
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
contents of referenced log file:
```0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Users\\MyUserName\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'clean' ]
2 info using npm@6.9.0
3 info using node@v10.13.0
4 verbose run-script [ 'preclean', 'clean', 'postclean' ]
5 info lifecycle spoke@2.0.0~preclean: spoke@2.0.0
6 info lifecycle spoke@2.0.0~clean: spoke@2.0.0
7 verbose lifecycle spoke@2.0.0~clean: unsafe-perm in lifecycle true
8 verbose lifecycle spoke@2.0.0~clean: PATH: C:\Users\MyUserName\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;c:\proj\MyProj\MyProj-2\node_modules\.bin;C:\Users\MyUser~1\AppData\Local\Temp\yarn--1561646646195-0.4332176644218666;c:\proj\MyProj\MyProj-2\node_modules\.bin;C:\Users\MyUserName\AppData\Local\Yarn\Data\link\node_modules\.bin;C:\Users\MyUserName\AppData\Local\Yarn\bin;C:\Program Files\libexec\lib\node_modules\npm\bin\node-gyp-bin;C:\Program Files\lib\node_modules\npm\bin\node-gyp-bin;C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin;C:\Python27\;C:\Python27\Scripts;C:\ProgramData\Boxstarter;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\iCLS\;C:\Program Files\Intel\Intel(R) Management Engine Components\iCLS\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\;C:\Program Files\Microsoft SQL Server\140\Tools\Binn\;C:\Program Files\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\ManagementStudio\;C:\Program Files\Git\cmd;C:\Program Files\TortoiseSVN\bin;C:\Program Files\TortoiseGit\bin;C:\Users\MyUserName\AppData\Local\Microsoft\WindowsApps;C:\Users\MyUserName\AppData\Roaming\npm;C:\Users\MyUserName\.dotnet\tools;C:\Users\MyUserName\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\MyUserName\AppData\Roaming\Dashlane\6.1907.0.17833\bin\Firefox_Extension\{442718d9-475e-452a-b3e1-fb1ee16b8e9f}\components;C:\Users\MyUserName\AppData\Roaming\Dashlane\6.1907.0.17833\ucrt;C:\Users\MyUserName\AppData\Roaming\Dashlane\6.1907.0.17833\bin\Qt;C:\Users\MyUserName\AppData\Roaming\Dashlane\6.1907.0.17833\bin\Ssl;C:\Users\MyUserName\AppData\Local\Programs\Fiddler;C:\Program Files\heroku\bin
9 verbose lifecycle spoke@2.0.0~clean: CWD: c:\proj\MyProj\MyProj-2
10 silly lifecycle spoke@2.0.0~clean: Args: [ '/d /s /c', 'rm -rf $OUTPUT_DIR' ]
11 silly lifecycle spoke@2.0.0~clean: Returned: code: 1  signal: null
12 info lifecycle spoke@2.0.0~clean: Failed to exec clean script
13 verbose stack Error: spoke@2.0.0 clean: 'rm -rf $OUTPUT_DIR'
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Users\MyUserName\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:182:13)
13 verbose stack     at ChildProcess.<anonymous> (C:\Users\MyUserName\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:182:13)
13 verbose stack     at maybeClose (internal/child_process.js:962:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
14 verbose pkgid spoke@2.0.0
15 verbose cwd c:\proj\MyProj\MyProj-2
16 verbose Windows_NT 10.0.17134
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\MyUserName\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "clean"
18 verbose node v10.13.0
19 verbose npm  v6.9.0
20 error code ELIFECYCLE
21 error errno 1
22 error spoke@2.0.0 clean: 'rm -rf $OUTPUT_DIR'
22 error Exit status 1
23 error Failed at the spoke@2.0.0 clean script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]+
```


**EDIT:**
Tried same w/ git bash since Procfile.dev had now been converted to LF and got the following error. Perhaps this will shed additional light on the issue.
```UserName@DESKTOP-AVQBLQO MINGW64 /c/source/myproj/myproj-2 (master)
$ yarn dev-start
yarn run v1.16.0
$ npm run clean

> spoke@2.0.0 clean C:\proj\MyProj\MyProj-2
> rm -rf $OUTPUT_DIR

$ nodemon -e js,jsx,md,mustache -w ./src --exec ./dev-tools/babel-run -- ./src/server
[nodemon] 1.18.9
[nodemon] reading config .\nodemon.json
[nodemon] to restart at any time, enter `rs`
[nodemon] or send SIGHUP to 14196 to restart
[nodemon] ignoring: .\src\client/**/* .\src\components/**/* .\src\containers/**/* .\src\store/**/* .\src\styles/**/* src/routes.jsx
[nodemon] watching: C:\proj\seiu\seiu-2\src/**/*
[nodemon] watching extensions: js,jsx,md,mustache
[nodemon] starting `./dev-tools/babel-run ./src/server`
[nodemon] spawning
[nodemon] child pid: 20964
'dev-tools\babel-run' is not recognized as an internal or external command,
operable program or batch file.
[nodemon] app crashed - waiting for file changes before starting...
[nodemon] watching 198 files```

**Comment by ibrand** [comment]
@groff0nthewall did you try my suggestion of re-cloning the repo from my branch to see if that fixed your problem? It might, and if it does, that change is probably going to be merged into the main branch soon.

**Comment by joshco** [comment]
If you are going to reclone...
I would highly recommend using Microsoft's new Ubuntu for Windows.  You'll have a real, binary compatible Ubuntu environment. It has made most windows specific nuances complicating my dev life go away.

https://www.microsoft.com/en-us/p/ubuntu/9nblggh4msv6?activetab=pivot:overviewtab

**Comment by groff0nthewall** [comment]
> @groff0nthewall did you try my suggestion of re-cloning the repo from my branch to see if that fixed your problem? It might, and if it does, that change is probably going to be merged into the main branch soon.

No, I was unable to clone. I got following error:
```
c:\proj\spoke>git clone -b fix-linefeed-issue git@github.com:MoveOnOrg/Spoke.git
Cloning into 'Spoke'...
The authenticity of host 'github.com (192.30.253.112)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'github.com,192.30.253.112' (RSA) to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by joshco** [comment]
@groff0nthewall you should clone the repo with the https url not the git one
do
> git clone -b fix-linefeed-issue __https://github.com/MoveOnOrg/Spoke.git__

not
> git clone -b fix-linefeed-issue git@github.com:MoveOnOrg/Spoke.git

**Comment by groff0nthewall** [comment]
> @groff0nthewall you should clone the repo with the https url not the git one
> do
> 
> > git clone -b fix-linefeed-issue **https://github.com/MoveOnOrg/Spoke.git**
> 
> not
> 
> > git clone -b fix-linefeed-issue [git@github.com](mailto:git@github.com):MoveOnOrg/Spoke.git

Thanks @joshco 
I copied/pasted from @ibrand's post wanting to follow the instructions verbatim. I will try your suggestion when I have a free moment.

**Comment by ibrand** [comment]
Ah sorry for misguiding you @groff0nthewall I'm realizing that I have ssh set up on my machine and forgot to change it for the instructions. Thanks @joshco !

**Comment by groff0nthewall** [comment]
Still no luck using the fix-linefeed-issue branch. And now I'm back to the ```node-pre-gyp``` error and cannot fix it this time with the --unsafe-perm flag as described here:  [mapbox/node-sqlite3#612](url)
```
$ yarn install
yarn install v1.16.0
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
info fsevents@1.2.7: The platform "win32" is incompatible with this module.
info "fsevents@1.2.7" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.4: The platform "win32" is incompatible with this module.
info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning "material-ui > react-event-listener@0.6.6" has incorrect peer dependency "react@^16.3.0".
[5/5] Building fresh packages...
[1/3] ⠂ uglifyjs-webpack-plugin
[2/3] ⠂ nodemon
error C:\proj\spoke-lf-fix\node_modules\sqlite3: Command failed.
Exit code: 1
Command: node-pre-gyp install --fallback-to-build
Arguments:
Directory: C:\proj\spoke-lf-fix\node_modules\sqlite3
Output:
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.6.38
node-pre-gyp info using node@10.13.0 | win32 | x64
node-pre-gyp info check checked for "C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64\node_sqlite3.node" (not found)
node-pre-gyp http GET https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.1.13/node-v64-win32-x64.tar.gz
node-pre-gyp http 403 https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.1.13/node-v64-win32-x64.tar.gz
node-pre-gyp ERR! Tried to download(403): https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.1.13/node-v64-win32-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for sqlite3@3.1.13 and node@10.13.0 (node-v64 ABI) (falling back to source compile with node-gyp)
node-pre-gyp http 403 status code downloading tarball https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.1.13/node-v64-win32-x64.tar.gz
node-pre-gyp ERR! Tried to download(undefined): https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.1.13/node-v64-win32-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for sqlite3@3.1.13 and node@10.13.0 (node-v64 ABI) (falling back to source compile with node-gyp)
node-pre-gyp http Connection closed while downloading tarball file

C:\proj\spoke-lf-fix\node_modules\sqlite3>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" clean )  else (node "" clean )

C:\proj\spoke-lf-fix\node_modules\sqlite3>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" clean )  else (node "" clean )
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@10.13.0 | win32 | x64
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@10.13.0 | win32 | x64
gyp info ok
gyp info ok

C:\proj\spoke-lf-fix\node_modules\sqlite3>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" configure --fallback-to-build --module=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64 )  else (node "" configure --fallback-to-build --module=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64 )

C:\proj\spoke-lf-fix\node_modules\sqlite3>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" configure --fallback-to-build --module=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64 )  else (node "" configure --fallback-to-build --module=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64 )
gyp info it worked if it ends withgyp ok
 info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@10.13.0 | win32 | x64
gyp info using node-gyp@3.8.0
gyp info using node@10.13.0 | win32 | x64
gyp info spawn C:\Python27\python.EXE
gyp info spawn args [ 'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=2015',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\proj\\spoke-lf-fix\\node_modules\\sqlite3\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\username\\.node-gyp\\10.13.0\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\username\\.node-gyp\\10.13.0',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\Users\\username\\.node-gyp\\10.13.0\\<(target_arch)\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\proj\\spoke-lf-fix\\node_modules\\sqlite3',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\proj\\spoke-lf-fix\\node_modules\\sqlite3\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn C:\Python27\python.EXE
gyp info spawn args [ 'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=2015',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\proj\\spoke-lf-fix\\node_modules\\sqlite3\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\username\\.node-gyp\\10.13.0\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\username\\.node-gyp\\10.13.0',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\Users\\username\\.node-gyp\\10.13.0\\<(target_arch)\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\proj\\spoke-lf-fix\\node_modules\\sqlite3',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\proj\\spoke-lf-fix\\node_modules\\sqlite3\\build',
gyp info spawn args   '-Goutput_dir=.' ]
Traceback (most recent call last):
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\gyp_main.py", line 16, in <module>
    sys.exit(gyp.script_main())
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 545, in script_main
    return main(sys.argv[1:])
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 538, in main
    return gyp_main(args)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 523, in gyp_main
    generator.GenerateOutput(flat_list, targets, data, params)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 2025, in GenerateOutput
    version=msvs_version)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\MSVSNew.py", line 213, in __init__
    self.Write()
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\MSVSNew.py", line 340, in Write
    f.close()
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\common.py", line 393, in close
    os.rename(self.tmp_path, filename)
WindowsError: [Error 183] Cannot create a file when that file already exists
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:345:16)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-build" "--module=C:\\proj\\spoke-lf-fix\\node_modules\\sqlite3\\lib\\binding\\node-v64-win32-x64\\node_sqlite3.node" "--module_name=node_sqlite3" "--module_path=C:\\proj\\spoke-lf-fix\\node_modules\\sqlite3\\lib\\binding\\node-v64-win32-x64"
gyp ERR! cwd C:\proj\spoke-lf-fix\node_modules\sqlite3
gyp ERR! node -v v10.13.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
gyp info ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'node-gyp.cmd configure --fallback-to-build --module=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (C:\proj\spoke-lf-fix\node_modules\sqlite3\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:962:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
node-pre-gyp ERR! System Windows_NT 10.0.17134
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\proj\\spoke-lf-fix\\node_modules\\sqlite3\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd C:\proj\spoke-lf-fix\node_modules\sqlite3
node-pre-gyp ERR! node -v v10.13.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.38
node-pre-gyp ERR! not ok
Failed to execute 'node-gyp.cmd configure --fallback-to-build --module=C:\proj\spoke-lf-fix\node_modules\sqlite3\lib\binding\node-v64-win32-x64\node_sqlite3.node --module_name
```

**Comment by ibrand** [comment]
@groff0nthewall did you ever figure anything out with this issue?

**Comment by groff0nthewall** [comment]
@ibrand  No, unfortunately, I was never able to resolve this issue. My client had to build and run for me on their servers every time I needed to test code changes. That contract has since ended and I no longer am working w/ Spoke at this time.

**Comment by ibrand** [comment]
Oof. Thank you for following up @groff0nthewall hopefully we can eventually get a working windows env...
